accepted/2025/direct-tool-execution.md

```diff
@@ -0,0 +1,114 @@
+
+## Detailed Design
+
+- When invoked, `dotnet tool exec` will:
+  1. Check if the specified tool is already installed locally and run it as a local tool
+  2. Check if the specified tool and version are present in the NuGet cache.
+  3. If not present, ask the user for permission download and install the tool package to the NuGet cache.
+    3a. No permission is requested if the tool is already installed locally or in the NuGet cache
+    3b. A command line parameter can provide the user approval for automated execution (ie -y / --yes)
+  4. Launch the tool with the provided arguments.
+- The tool is not added to any manifest and is not available globally or locally after execution.
+- The command supports passing all arguments after `--` directly to the tool.
+- Tools executed via `dotnet tool exec` are always installed to the NuGet cache, leveraging existing NuGet cache policies and avoiding the need for a manifest. This simplifies the process and ensures ephemeral tool usage.
+
+## Considerations
+
+- The tool is ephemeral and not discoverable via `dotnet tool list`.
+- Caching behavior follows standard NuGet cache policies.
+- There should be minimal impact to the state of the machine beyond the install to the NuGet cache.
+
+### Calling by Command vs. Package
+
+There is the potential of supporting calling a tool by its command name rather than the package ID. This approach could improve usability, but it is potentially problematic because .NET tools do not have a standard naming convention for commands, and there are existing protections on nuget.org to guard against spelling errors and package squatting that apply only to package IDs. Further discussion and feedback are needed to determine the best approach for this scenario.
+
+We could support a standard naming convention for the package like "dotnet-<command>" as that's used in dotnet-ef and many other tools but that would open the question of what to do about existing tools.
+
+---
+
+## `dnx` Summary
+
+This proposal introduces a new launcher script, `dnx`, which acts as a thin wrapper for `dotnet tool exec`. The script will be installed as `dnx.cmd` on Windows and as an executable `dnx` (no extension) on non-Windows platforms, and will reside alongside the `dotnet` host executable.
+
+When invoked, the `dnx` script will forward all arguments to a .NET CLI command `dotnet dnx` which will initially just be an alias for `dotnet tool exec`, enabling a familiar and concise entry point for running .NET tools directly. The script will not contain any additional logic, ensuring consistency and simplifying maintenance across platforms.
+
+## Motivation
+
+The `dnx` command provides a short, memorable alias for running .NET tools, inspired by the historical DNX (DotNet Execution Environment) command. It simplifies tool invocation, especially in cross-platform scripts and documentation, and provides a consistent experience across operating systems.
```

Comment:
I think it's worth noting that using `dnx` for this is also inspired by `npx`, and the similarity is helpful since many devs are already familiar with `npx`.